src/main/scala/mesosphere/marathon/core/task/termination/impl/TaskKillServiceActor.scala

```diff
@@ -168,3 +175,31 @@ private[termination] object TaskKillServiceActor {
     clock: Clock): Props = Props(
     new TaskKillServiceActor(taskTracker, driverHolder, stateOpProcessor, config, clock))
 }
+
+/**
+  * Wraps a timer into an interface that hides internal mutable state behind simple setup and cancel methods
+  */
+private[this] trait RetryTimer {
+  private[this] var retryTimer: Option[Cancellable] = None
+
+  /** Creates a new timer when setup() is called */
+  def createTimer: () => Cancellable
```

Comment:
createTimer is  a function without parentheses that gives you a function of arity 0.
I would argue, defining it simply as:
```scala
def createTimer(): Cancellable
```
would be more natural and easy to read here.